### PR TITLE
lftp 4.7.2

### PR DESCRIPTION
**Pull request by nijikon**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

### CODE REVIEW

File: Formula/lftp.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 class Lftp < Formula
   desc "Sophisticated file transfer program"
   homepage "https://lftp.yar.ru/"
-  url "https://lftp.yar.ru/ftp/lftp-4.6.5.tar.xz"
```
Comment (DomT4) [inline Formula/lftp.rb]: Did upstream lose https support?

**Comment by DomT4** [comment]
Failing to build on all platforms:

```
/bin/sh ../libtool --silent  --tag=CXX   --mode=compile clang++ -DHAVE_CONFIG_H -I. -I../lib  -I../lib -I../trio -I/usr/include -I/usr/local/opt/openssl/include    -O2 -Wall -Wwrite-strings -Woverloaded-virtual -fno-exceptions -fno-rtti -fno-implement-inlines -c -o liblftp_network_la-RateLimit.lo `test -f 'RateLimit.cc' || echo './'`RateLimit.cc
lftp_ssl.cc:827:14: error: use of undeclared identifier 'SSL_OP_NO_TLSv1_1'
         {"-TLS1.1",SSL_OP_NO_TLSv1_1},
                    ^
lftp_ssl.cc:828:14: error: use of undeclared identifier 'SSL_OP_NO_TLSv1_2'
         {"-TLS1.2",SSL_OP_NO_TLSv1_2},
                    ^
```

I don't even get that far locally, for some reason:

```
clang -DHAVE_CONFIG_H -I. -I../lib  -I../lib -I../trio -I/usr/include/readline   -g -O2 -Wall -c -o lftp-lftp_rl.o `test -f 'lftp_rl.c' || echo './'`lftp_rl.c
In file included from lftp_rl.c:23:
In file included from ./history.h:23:
In file included from ./keyvalue.h:23:
In file included from ./xstring.h:97:
./xmalloc.h:31:38: error: C does not support default arguments
char *xstrdup(const char *s,int spare=0);
                                     ^~
./xmalloc.h:32:21: error: expected ')'
char *xstrset(char *&mem,const char *s);
                    ^
./xmalloc.h:32:14: note: to match this '('
char *xstrset(char *&mem,const char *s);
             ^
./xmalloc.h:33:21: error: expected ')'
char *xstrset(char *&mem,const char *s,size_t n);
                    ^
./xmalloc.h:33:14: note: to match this '('
char *xstrset(char *&mem,const char *s,size_t n);
             ^
In file included from lftp_rl.c:23:
In file included from ./history.h:23:
In file included from ./keyvalue.h:23:
./xstring.h:104:1: error: unknown type name 'class'
class xstring0 // base class
^
./xstring.h:104:15: error: expected ';' after top level declarator
class xstring0 // base class
              ^
              ;
./xstring.h:123:1: error: unknown type name 'class'
class xstring_c : public xstring0
^
./xstring.h:123:17: error: expected ';' after top level declarator
class xstring_c : public xstring0
                ^
                ;
./xstring.h:123:19: error: unknown type name 'public'
class xstring_c : public xstring0
                  ^
./xstring.h:123:34: error: expected ';' after top level declarator
class xstring_c : public xstring0
                                 ^
                                 ;
./xstring.h:157:1: error: unknown type name 'class'
class xstring_ca : public xstring_c
^
./xstring.h:157:18: error: expected ';' after top level declarator
class xstring_ca : public xstring_c
                 ^
                 ;
./xstring.h:157:20: error: unknown type name 'public'
class xstring_ca : public xstring_c
                   ^
./xstring.h:157:36: error: expected ';' after top level declarator
class xstring_ca : public xstring_c
                                   ^
                                   ;
./xstring.h:165:1: error: unknown type name 'class'
class xstring_clonable;
^
./xstring.h:167:1: error: unknown type name 'class'
class xstring : public xstring0
^
./xstring.h:167:15: error: expected ';' after top level declarator
class xstring : public xstring0
              ^
              ;
./xstring.h:167:17: error: unknown type name 'public'
class xstring : public xstring0
                ^
./xstring.h:167:32: error: expected ';' after top level declarator
class xstring : public xstring0
                               ^
                               ;
./xstring.h:286:1: error: unknown type name 'class'
class xstring_clonable : public xstring {};
^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [lftp-lftp_rl.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [install-recursive] Error 1
make: *** [install] Error 2
```

**Comment by nijikon** [comment]
More errors
```
clang++ -DHAVE_CONFIG_H -I. -I../lib  -I../lib -I../trio -I/usr/local/opt/readline/include/readline   -O2 -Wall -Wwrite-strings -Woverloaded-virtual -fno-exceptions -fno-rtti -fno-implement-inlines -c -o lftp-lftp_rl.o `test -f 'lftp_rl.cc' || echo './'`lftp_rl.cc
lftp_rl.cc:32:4: error: unknown type name 'HIST_ENTRY'
   HIST_ENTRY *temp;
   ^
lftp_rl.cc:36:4: error: use of undeclared identifier 'using_history'
   using_history();
   ^
lftp_rl.cc:37:9: error: use of undeclared identifier 'previous_history'
   temp=previous_history();
        ^
lftp_rl.cc:39:7: error: use of undeclared identifier 'add_history'
      add_history(cmd_buf);
      ^
lftp_rl.cc:41:4: error: use of undeclared identifier 'add_history_time'
   add_history_time(ts);
   ^
lftp_rl.cc:42:4: error: use of undeclared identifier 'using_history'
   using_history();
   ^
lftp_rl.cc:57:11: error: use of undeclared identifier 'history_expand'
   return history_expand((char*)what,where);
          ^
lftp_rl.cc:62:4: error: use of undeclared identifier 'using_history'
   using_history();
   ^
lftp_rl.cc:63:11: error: use of undeclared identifier 'read_history'
   return read_history(fn);
          ^
lftp_rl.cc:68:4: error: use of undeclared identifier 'using_history'
   using_history();
   ^
lftp_rl.cc:69:11: error: use of undeclared identifier 'write_history'
   return write_history(fn);
          ^
lftp_rl.cc:74:4: error: unknown type name 'HISTORY_STATE'
   HISTORY_STATE *st = history_get_history_state();
   ^
lftp_rl.cc:74:24: error: use of undeclared identifier 'history_get_history_state'
   HISTORY_STATE *st = history_get_history_state();
                       ^
lftp_rl.cc:75:4: error: unknown type name 'HIST_ENTRY'
   HIST_ENTRY *hist;
   ^
lftp_rl.cc:77:4: error: use of undeclared identifier 'using_history'
   using_history();
   ^
lftp_rl.cc:79:8: error: use of undeclared identifier 'history_base'
   i = history_base + st->length - cnt;
       ^
lftp_rl.cc:80:24: error: use of undeclared identifier 'history_base'
   if(cnt == -1 || i < history_base) i = history_base;
                       ^
lftp_rl.cc:80:42: error: use of undeclared identifier 'history_base'
   if(cnt == -1 || i < history_base) i = history_base;
                                         ^
lftp_rl.cc:83:18: error: use of undeclared identifier 'history_get'
   while((hist = history_get(i))) {
                 ^
clang++ -DHAVE_CONFIG_H -I. -I../lib  -I../lib -I../trio -I/usr/local/opt/readline/include/readline   -O2 -Wall -Wwrite-strings -Woverloaded-virtual -fno-exceptions -fno-rtti -fno-implement-inlines -c -o lftp-attach.o `test -f 'attach.cc' || echo './'`attach.cc
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make[2]: *** [lftp-lftp_rl.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [install-recursive] Error 1
make: *** [install] Error 2```

**Comment by nijikon** [comment]
@DomT4 please review

File: Formula/lftp.rb [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
   end
 
+  depends_on "automake" => :build
+  depends_on "autoconf" => :build
   depends_on "pkg-config" => :build
   depends_on "readline"
-  depends_on "openssl"
+  depends_on "openssl" if MacOS.version < :yosemite
+
+  patch do
```
Comment (DomT4) [inline Formula/lftp.rb]: Same question re has this been submitted upstream/etc?

File: Formula/lftp.rb [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
     sha256 "182979ca79b7ea9d9e70e9fefab990d5d238da709ce0e17edf5ec7166b976fb1" => :yosemite
     sha256 "dc44f63a8f1767af2b4ed2008c199377eabd117339f01ee17bb73aa844cf3dee" => :mavericks
   end
 
+  depends_on "automake" => :build
+  depends_on "autoconf" => :build
   depends_on "pkg-config" => :build
   depends_on "readline"
-  depends_on "openssl"
+  depends_on "openssl" if MacOS.version < :yosemite
```
Comment (DomT4) [inline Formula/lftp.rb]: Is this using CommonCrypto on newer systems now?

Comment (nijikon) [inline Formula/lftp.rb]: I don't know

File: Formula/lftp.rb (hunk 2 above)
Comment (nijikon) [inline Formula/lftp.rb]: It's reported in the linked PR and all patches are from there.

File: Formula/lftp.rb (hunk 3 above)
Comment (DomT4) [inline Formula/lftp.rb]: Looking at the upstream repo it _seems_ to use either OpenSSL or GnuTLS, so not sure we should be limiting this to `:yosemite`. Can you check the linkage of this PR with `depends_on "openssl"` to see if it's actually linking?

File: Formula/lftp.rb (hunk 2 above)
Comment (DomT4) [inline Formula/lftp.rb]: Please note that link in the formula with the standard `#`. We can't realistically expect future contributors to comb through old PRs for answers.

File: Formula/lftp.rb (hunk 3 above)
Comment (apjanke) [inline Formula/lftp.rb]: I ran it on El Cap and it's still trying to use `openssl`, which is now an error when it tries to use the system `openssl`. That means this dependency should not be limited to `< :yosemite`.

```

In file included from FileAccess.cc:1021:
In file included from ./ftpclass.h:29:
./lftp_ssl.h:27:12: fatal error: 'openssl/ssl.h' file not found
#  include <openssl/ssl.h>
           ^
/bin/sh ../libtool --silent  --tag=CXX   --mode=compile clang++ -DHAVE_CONFIG_H -I. -I../lib  -I../lib -I../trio   -O2 -Wall -Wwrite-strings -Woverloaded-virtual -fno-exceptions -fno-rtti -fno-implement-inlines -c -o StringSet.lo StringSet.cc
1 error generated.
make[2]: *** [FileAccess.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [install-recursive] Error 1
make: *** [install] Error 2
```

**Comment by apjanke** [comment]
Build is failing on Yosemite only when `depends_on 'openssl'` is included (up in #4140).

The author says:

> I don't have OS X anywhere nearby. Sorry.

and there's no indication that the patches for OS X are going to make it in to upstream. Maybe this is a candidate for boneyarding until someone makes an OS X port of `lftp`, or at least skipping this version?

**Comment by dunn** [comment]
Yeah, I'd say skip this version and unless the next version works, 💀

**Comment by DomT4** [comment]
FWIW, `4.6.6` builds so it might be worth updating to that at least, but I hadn't managed to get any of the new 4.7.x branch building to date.

**Comment by MoOx** [comment]
FYI 4.6.6 seems to be unavailable or something


```
❯❯ brew install homebrew/boneyard/lftp

==> Installing lftp from homebrew/boneyard
==> Downloading https://lftp.yar.ru/ftp/lftp-4.6.6.tar.xz

curl: (7) Failed to connect to lftp.yar.ru port 443: Connection refused
Error: Failed to download resource "lftp"
Download failed: https://lftp.yar.ru/ftp/lftp-4.6.6.tar.xz

```

I tried with a VPN, same issue, and I can't find the link to 4.6.6 on the website. So this sucks :(

**Comment by MoOx** [comment]
Workaround is to get version from here ftp://ftp.st.ryukoku.ac.jp/pub/network/ftp/lftp/ and move it to brew cache in ~/Library/Caches/Homebrew/

**Comment by tuupola** [comment]
Currently downloads ok.

```
$ wget https://lftp.yar.ru/ftp/lftp-4.6.6.tar.xz
--2016-10-14 16:42:47--  https://lftp.yar.ru/ftp/lftp-4.6.6.tar.xz
Resolving lftp.yar.ru... 213.187.99.145, 2a03:ca00:2:908:a00:27ff:fe6d:a9cc
Connecting to lftp.yar.ru|213.187.99.145|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1518544 (1.4M) [application/x-xz]
Saving to: 'lftp-4.6.6.tar.xz'

lftp-4.6.6.tar.xz   100%[=====================>]   1.45M  2.27MB/s   in 0.6s   

2016-10-14 16:42:49 (2.27 MB/s) - 'lftp-4.6.6.tar.xz' saved [1518544/1518544]
```

**Comment by registe** [comment]
Try this
https://github.com/lavv17/lftp/releases
http://lftp.tech/get.html

**Comment by dunn** [comment]
See https://github.com/Homebrew/homebrew-core/pull/7371

**Pull request closed without merging**